gandalf.js

```diff
@@ -3,36 +3,40 @@
  *
```

Comment:
SCRIPT-Nice job on the comprehensive test coverage, it gives a lot of confidence in the stability of the new feature.